### ISSUE TITLE
Consider using PanoMarkers for label drawing layer?

### ISSUE THREAD

**Issue by aileenzeng** [ProjectSidewalk/SidewalkWebpage#1287]
Google Maps has a PanoMarker add-on that allows you to place a marker on a panorama and will automatically adjust its location as the user changes their heading/pitch/zoom. It's also possible to add listeners and customize the PanoMarker. Currently, we're doing all the math of manually to adjust label positions, so we have a ton of different functions to take care of rendering. Using PanoMarkers instead could probably help simplify our code.

Example of PanoMarkers in action:
http://marmat.github.io/google-maps-api-addons/panomarker/examples/fancy.html
![image](https://user-images.githubusercontent.com/25534091/42913415-bdb3499a-8aa9-11e8-8a96-547a5a51d0b5.png)

**Comment by misaugstad** [comment]
I remember trying to use PanoMarkers for something related to PS awhile ago, and there was some major problem with them, but I don't remember exactly what it was... Was it that you could only use lat-lngs to add them to a pano or something? Did you run into any problems with this method @aileenzeng?

**Comment by jonfroehlich** [comment]
Yes, I think when @aileenzeng wrote this Issue--with my encouragement--she hadn't yet fully realized how and why we had to write our own custom code for placing a label on the pano. @aileenzeng, can you explain now why this won't work? I remember you had to abandon this approach for your own labeling tool (the qualitative review one) but I can't remember why.

**Comment by aileenzeng** [comment]
I ended up actually being able to use PanoMarkers for the label viewing panorama tool (and verified that it can place labels onto the screen accurately). The location of a PanoMarker is set using heading/pitch values. We have some functions that allow us to calculate these accurately from the label's x-y coordinates, screen dimensions and original heading/pitch/zoom values. One of the bugs that I found with PanoMarkers is that if you zoom in to the highest zoom value, it will 'jump' to a slightly new location (but will go back to its correct position once you zoom out). 

The tool that I was having trouble with earlier are the official Google Maps markers -- these only accept lat-lng coordinates (rather than heading/pitch) for position.

**Comment by misaugstad** [comment]
@aileenzeng having worked with PanoMarkers quite a bit now, do you think it is feasible to swap out our current implementation for PanoMarkers without losing any functionality? Furthermore, do you think this would help solve our issue of not being able to resize the GSV window?

**Comment by jonfroehlich** [comment]
@tongning was also working on this so should be involved in this conversation imo.

**Comment by jonfroehlich** [comment]
Since @jsomeara has recently been working on something related (sticky labels), perhaps worth him weighing in on this? And maybe we close this old ticket?

Related to: https://github.com/ProjectSidewalk/SidewalkWebpage/issues/3353

**Comment by jsomeara** [comment]
Pano markers might help simplify code, but imo it's not worth it because we sacrifice a huge amount of control and customizability.

**Comment by misaugstad** [comment]
Sounds good. If we ever do a full rewrite of the Explore page labeling code we can always revisit. Closing for now.
